spec/integration/application/apply_spec.rb

```diff
@@ -460,9 +460,10 @@ def bogus()
     let(:env) { Puppet::Node::Environment.create(env_name.to_sym, [File.join(populated_env_dir, 'spec', 'modules')]) }
     let(:node) { Puppet::Node.new('test', :environment => env) }
 
-    around(:each) do |example|
+    before(:each) do
+
       Puppet[:rich_data] = rich_data
-      Puppet.override(:loaders => Puppet::Pops::Loaders.new(env)) { example.run }
+      Puppet.push_context(:loaders => Puppet::Pops::Loaders.new(env))
```

Comment:
One way to ensure the stack remains balanced is to do:

```ruby
around(:each) do |example|
  Puppet.push_context(:loaders => Puppet::Pops::Loaders.new(env))
  example.run
ensure
  Puppet.pop_context
end
```

Reply:
I couldn't find a way to ensure the `before(:each)` block defined in the spec_helper.rb executed prior to the `:around` blocks; this resulted in many tests failing with the `Puppet[:libdir]` code failing because `Puppet[:vardir]` was undefined. The only way that I could was to update all of the tests to stop using `:around`.